project-reports/2023/2023-Q2-Hyperledger-Caliper.md

```diff
@@ -0,0 +1,57 @@
+---
+layout: default
+title: 2023 Q2 Hyperledger Caliper
+parent: 2023
+grand_parent: Project Updates
+---
+
+# Project Health
+
+* 2 active maintainers
+* PRs continuously submitted and merged (non-bot PR rate dropped a bit since the previous quarter, expected to improve in the next quarter due to mentorship-related updates).
+* The PR backlog increased a bit due to version bumping bots.
+* Questions on Discord and GitHub are regularly answered.
+* Monthly contributor calls (discussion happens mostly on Discord, as needed).
+* Mentorship-related preparations and improvements are expected in the next quarter.
+
+# Questions/Issues for the TOC
+
+Not at this time.
+
+# Releases
+
+The latest stable release is v0.5.0; we're also continuously publishing unstable releases after every merged PR:
+
+* [npm](https://www.npmjs.com/package/@hyperledger/caliper-cli/v/0.5.0)
+* [DockerHub](https://hub.docker.com/layers/caliper/hyperledger/caliper/0.5.0/images/sha256-3a7df18ef6a17a8c851dac19241049be2375dc892643c576932f360ca9d7198c?context=explore)
+
+# Overall Activity in the Past Quarter
+
+[List of merged PRs (1)](https://github.com/hyperledger/caliper/pulls?q=is%3Apr+merged%3A2023-03-28..2023-06-26+)
+
+Summary of activities:
+* Docker monitoring bug fix
+* Evaluating Caliper capabilities/limitations for upcoming mentorship project
+
+The activity has been minimal in the last quarter due to the maintainers being otherwise engaged. 
```

Comment:
Do you have a roadmap for Caliper and goals for what you want to accomplish in the next year?

Reply:
The Fabric integration and core features are stable enough to conduct large scale experiments in terms of workload generation. There hasn't been much interest in other integrations recently that would warrant the improvement and maintenance of other platform connectors. So currently we mainly provide support for the community regarding the usage of Caliper (configuration, best practices), rarely about missing features. 

There are ongoing mentorships that will utilize Caliper in some way, so serving their needs will be the main concern for the next months (probably bringing some improvements to Caliper too). After that, some technical debts should be resolved (tests, documentation, tutorials, workshop material), and hopefully that'll push Caliper into a stable/major release state.

There are ongoing discussions about a performance task force (or some other form), so we'll see what role Caliper will play in that. That are some tooling around blockchain performance that could be consolidated too, but that'll be a bit more complicated work/discussion I think :)

Reply:
If the project maintainers are willing to, there can be ambitious goals that benefits both the Hyperledger Foundation, user base of the bigger Hyperledger projects and the Caliper project itself. Following are the ways I would think Caliper can be a value add
1. Either the Caliper team or the task force can come up with a list of variable parameters, and their influence in the throughput of blockchain commits.
2. Caliper team can define a means of certifying a blockchain deployment platform through benchmarking capabilities. This can be done through a combination of pre-defined test cases and required deployment structure. I would visualize this as a means of compliance suite for the blockchain deployment platforms so that they are aware of complex and otherwise overlooked scenarios that are possible.